### PR TITLE
feat(backend): add customer to backend

### PR DESCRIPTION
**Pull request by PascalHelbig**

### CODE REVIEW

File: app/backend/graphql/customer/mutations/customerDeleteMutationField.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  type: Customer,
+  description: 'Delete a customer',
+  args: {
+    customerId: idArg({ description: 'Id of the customer' }),
+  },
+  authorize: async (_source, { customerId }, context) => {
+    const customer = await context.prisma.customer.findUnique({ where: { id: customerId }, rejectOnNotFound: true })
+    return isTeamMember({ id: customer.teamId }, context)
+  },
+  resolve: (_source, { customerId }, context) => context.prisma.customer.delete({ where: { id: customerId } }),
```
Comment (moncapitaine) [inline app/backend/graphql/customer/mutations/customerDeleteMutationField.ts:15]: We need to be sure that no project with time- or invoice data gets deleted accidentally

File: app/backend/graphql/customer/queries/customerQueryField.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { idArg, queryField } from 'nexus'
+import { isTeamMember } from '../../team/utils'
+import { Customer } from '../customer'
+
+export const customerQueryField = queryField('customer', {
```
Comment (moncapitaine) [inline app/backend/graphql/customer/queries/customerQueryField.ts:5]: This just gets the title for a customer with a given id. What we need is a query that returns all customers for a given team but I guess this is part of the team object.

File: app/backend/graphql/team/team.ts [hunk 3]
```diff
@@ -21,5 +22,10 @@ export const Team = objectType({
           where: { teamMemberships: { some: { teamId: team.id } } },
         }),
     })
+    t.list.field('customers', {
```
Comment (moncapitaine) [inline app/backend/graphql/team/team.ts:25]: here we are, you can cancel my comment above.

**Review by moncapitaine — APPROVED** [review]

File: app/backend/graphql/customer/mutations/customerDeleteMutationField.ts (hunk 1 above)
Comment (PascalHelbig) [reply app/backend/graphql/customer/mutations/customerDeleteMutationField.ts:15]: Let's look at that again when we link projects and invoices to the customer.

**Pull request merged**